crates/pipeline_manager/src/api/pipeline.rs

```diff
@@ -7,604 +17,481 @@ use actix_web::{
     web::{self, Data as WebData, ReqData},
     HttpRequest, HttpResponse,
 };
+use chrono::{DateTime, Utc};
 use log::{debug, info};
-use pipeline_types::config::RuntimeConfig;
+use pipeline_types::config::{PipelineConfig, RuntimeConfig};
+use pipeline_types::error::ErrorResponse;
 use serde::{Deserialize, Serialize};
 use utoipa::{IntoParams, ToSchema};
-
-use crate::{
-    api::{examples, parse_string_param},
-    auth::TenantId,
-    db::{storage::Storage, AttachedConnector, DBError, PipelineId, Version},
-};
-
-use super::{ManagerError, ServerState};
 use uuid::Uuid;
 
-/// Request to create a new pipeline.
-#[derive(Debug, Deserialize, ToSchema)]
-#[serde(deny_unknown_fields)]
-pub(crate) struct NewPipelineRequest {
-    /// Unique pipeline name.
-    name: String,
-    /// Pipeline description.
-    description: String,
-    /// Name of the program to create a pipeline for.
-    program_name: Option<String>,
-    /// Pipeline configuration parameters (e.g. number of workers).
-    /// These knobs are independent of any connector attached to the pipeline.
-    config: RuntimeConfig,
-    /// Attached connectors.
-    connectors: Option<Vec<AttachedConnector>>,
-}
-
-/// Response to a pipeline creation request.
-#[derive(Serialize, ToSchema)]
-pub(crate) struct NewPipelineResponse {
-    /// ID of the newly created pipeline.
-    pipeline_id: PipelineId,
-    /// Initial pipeline version (this field is always set to 1).
-    version: Version,
+// REGULAR ENDPOINTS
+
+/// Extended pipeline descriptor with code being optionally included.
```

Comment:
maybe it should just always have the code?

Reply:
It would simplify it though -- let's see in terms of performance how we fare